model/historical_test_data.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/pkg/errors"
+)
+
+// HistoricalTestData describes aggregated test result data for a given date
+// range.
+type HistoricalTestData struct {
+	Info            HistoricalTestDataInfo
+	NumPass         int       `bson:"num_pass"`
+	NumFail         int       `bson:"num_fail"`
+	AverageDuration float64   `bson:"average_duration"`
+	LastUpdate      time.Time `bson:"last_update"`
+
+	env       cedar.Environment
+	populated bool
+}
+
+// CreateHistoricalTestData is an entry point for creating a new
+// HistoricalTestData.
+func CreateHistoricalTestData(info HistoricalTestDataInfo) (*HistoricalTestData, error) {
+	if err := info.validate(); err != nil {
+		return nil, err
+	}
+
+	return &HistoricalTestData{
+		Info:       info,
+		LastUpdate: time.Now(),
+		populated:  true,
+	}, nil
+}
+
+// HistoricalTestDataInfo describes information unique to a single test
+// statistics document.
+type HistoricalTestDataInfo struct {
+	Project   string    `bson:"project"`
+	Variant   string    `bson:"variant"`
+	TaskName  string    `bson:"task_name"`
+	TestName  string    `bson:"test_name"`
+	Requester string    `bson:"requester"`
```

Comment:
requester is a weird evergreen term that maybe we want to elide/cover up?

Reply:
its used as part of the identity... would you rather call it something else?

Reply:
I don't think we can get rid of it though

Reply:
We should definitely call it something else, and maybe have a maineline boolean in addition to  some better kind of name.

Reply:
What is a good substitute? `request_type`? Historical test statistics uses requester as part of the id and its not binary (like mainline), so we need to keep an equivalent field I think.